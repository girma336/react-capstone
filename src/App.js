import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Route, Routes, BrowserRouter } from 'react-router-dom';
import { feachData } from './Redux/elementSlices';
import SearchElement from './component/SearchElement';
import DetailElement from './component/detailElement';

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(feachData());
  });

  return (
    <BrowserRouter>
      <div>
        <Routes>
          <Route path="/" element={<SearchElement />} />
          <Route path="/detailes/:idAPP" element={<DetailElement />} />
        </Routes>
      </div>
    </BrowserRouter>
  );
};

export default App;
