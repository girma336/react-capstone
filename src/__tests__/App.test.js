import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../Redux/configurStore';

describe('test App', () => {
  it('test snapshoot of app', () => {
    const tree = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
