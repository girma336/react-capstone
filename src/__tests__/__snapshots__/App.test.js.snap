// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App test snapshoot of app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <header>
            <div
              class="header-logo"
            >
              <div>
                <h1>
                  Final Space Character
                </h1>
              </div>
            </div>
            <div
              class="list-header"
            >
              <div
                class="search field is-horizontal"
              >
                <input
                  class="input"
                  value=""
                />
                <svg
                  class="label-1"
                  fill="currentColor"
                  height="1em"
                  id="search"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="1em"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g>
                    <path
                      d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </header>
          <div>
            <div
              class="container"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <header>
          <div
            class="header-logo"
          >
            <div>
              <h1>
                Final Space Character
              </h1>
            </div>
          </div>
          <div
            class="list-header"
          >
            <div
              class="search field is-horizontal"
            >
              <input
                class="input"
                value=""
              />
              <svg
                class="label-1"
                fill="currentColor"
                height="1em"
                id="search"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 24 24"
                width="1em"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <g>
                  <path
                    d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </header>
        <div>
          <div
            class="container"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
